webapp/tests/Controller/API/BaseTest.php

```diff
@@ -61,4 +83,165 @@ protected function verifyApiJsonResponse(
         self::assertEquals($status, $response->getStatusCode(), $message);
         return json_decode($response->getContent(), true);
     }
+
+    /**
+     * Test that the list action returns the expected objects
+     */
+    public function testList()
+    {
+        if ($this->apiEndpoint === null) {
+            static::markTestSkipped('No endpoint defined');
+        }
+        $objects = $this->verifyApiJsonResponse('GET', "/contests/2/$this->apiEndpoint", 200, $this->apiUser);
+
+        static::assertIsArray($objects);
+        foreach ($this->expectedObjects as $expectedObjectId => $expectedObject) {
+            $object = null;
+            foreach ($objects as $potentialObject) {
+                if ($potentialObject['id'] == $expectedObjectId) {
+                    $object = $potentialObject;
+                    break;
+                }
+            }
+
+            static::assertNotNull($object);
+            foreach ($expectedObject as $key => $value) {
+                static::assertEquals($value, $object[$key]);
+            }
+        }
+
+        foreach ($this->expectedAbsent as $expectedAbsentId) {
+            $object = null;
+            foreach ($objects as $potentialObject) {
+                if ($potentialObject['id'] == $expectedAbsentId) {
+                    $object = $potentialObject;
+                    break;
+                }
+            }
+
+            static::assertNull($object);
+        }
+    }
+
+    /**
+     * Test that the list action returns the expected objects if ID's are passed
+     */
+    public function testListWithIds()
+    {
+        if ($this->apiEndpoint === null) {
+            static::markTestSkipped('No endpoint defined');
+        }
+        $objects = $this->verifyApiJsonResponse('GET', "/contests/2/$this->apiEndpoint?" . http_build_query(['ids' => array_keys($this->expectedObjects)]), 200, $this->apiUser);
```

Comment:
What do you propose/? I can put it in a protected property in this class? Or load the last contest? But the latter also seems ugly.

Reply:
Why is it 2 here and 3 down below? This looks brittle. Do we have a better way to identify it?

Reply:
I can load the 2 on startup by loading the contest with shortname `demo`, since that is what it is.
The 3 is a contest dat doesn't exist (on purpose). I could change the ID to `nonexistent` to make that clear?

Reply:
I would keep the ID numeric. What about increasing the ID, so that it doesn't collide even if we later decide to add more contests, e.g. `42` and then a comment there that we don't expect it to exist.

Reply:
Done